### ISSUE TITLE
Loading solely from file system is very hard

### ISSUE THREAD

**Issue by JakeWharton** [cashapp/zipline#1380]
I want to load from the file system in tests. This is very had to do. Everything about the current design of loader assumes remote loading.

I'm not sure if the current design can even be adapted to accommodate this case, but perhaps a new foundational design could allow a more modular composition of load sources.

**Comment by swankjesse** [comment]
Agreed about a new foundational design. ZiplineLoader has some ideas that haven’t worked out very well in practice, and I’d like to consider building a replacement.

I think our design is too imperative. I’d like to change it to make a loading decision, then execute it, as I describe here:
https://publicobject.com/2024/03/10/deciders-decisions-and-executors/

There’s a lot of moving parts to the decision and ideally we’re extremely conservative about how much I/O we do.
